src/lib/y2partitioner/actions/delete_bcache.rb

```diff
@@ -0,0 +1,97 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2partitioner/actions/delete_device"
+
+module Y2Partitioner
+  module Actions
+    # Action for deleting a Bcache
+    #
+    # @see DeleteDevice
+    class DeleteBcache < DeleteDevice
+      def initialize(*args)
+        super
+        textdomain "storage"
+      end
+
+    private
+
+      # Deletes the indicated Bcache (see {DeleteDevice#device})
+      def delete
+        log.info "deleting bcache raid #{device}"
+        device_graph.remove_bcache(device)
+      end
+
+      # Confirmation before performing the delete action
+      #
+      # @return [Boolean]
+      def confirm
+        if device.partitions.any?
```

Comment:
I am not sure how to react if whole bcache device is used as source for something else like LVM or md raid. Should I check that there is no blk devices in children or predecessors of device?

Reply:
Why would we need to check the predecessors?

Checking whether it has descendants and asking for confirmation about deleting those should be enough. By the way, that's something that I kind of expected to see in the base class `Actions::DeleteDevice`

Reply:
yeah, me too. I am not sure if check for any descendants as it can be only empty partition_table or fs, which is probably fine, not?